### ISSUE TITLE
[HASKELL] Incorrect escaping when generating field names

### ISSUE THREAD

**Issue by mmosemmle** [swagger-api/swagger-codegen#5793]
##### Description
The 'Type mismatch on container access' issues found by lgtm.com:
https://lgtm.com/projects/g/swagger-api/swagger-codegen/alerts/
means that property names are not properly escaped when haskell code is generated.

##### Swagger-codegen version
master @ec19e8f

##### Swagger declaration file content or url
https://gist.github.com/mmosemmle/c2165d6eb37e77b670c383b37118666d

##### Command line used for generation
```
java -jar swagger-codegen/modules/swagger-codegen-cli/target/swagger-codegen-cli.jar generate -l haskell -i swagger.yaml -o ./test
```
##### Steps to reproduce
Download the swagger.yaml file from the above link, then do use the above command, and check
./test/lib/TestGeneratedBug/Type.hs
to see a field called Test??-Data in line 22.

##### Suggest a Fix

The fix is fairly simple and I am happy to create a PR to fix it. Thanks.

**Comment by wing328** [comment]
@mmosemmle thanks for reporting the issue and creating a PR for the fix.

Let me know if you need help with the PR.

**Comment by mmosemmle** [comment]
Thanks. Created a PR for the fix. Btw, it may also be useful to try out the PR integration from lgtm.com, which will discover issues like this during a PR and avoid them being merged. Thanks.

**Comment by wing328** [comment]
Closed via #5794 

Thanks for the suggestion about lgtm.com. Will check it out later.
